server/controllers/assessmentDayDataController/index.js

```diff
@@ -0,0 +1,66 @@
+import AssessmentDayDataModel from '../../models/AssessmentDayDataModel'
+import DataFileModel from '../../models/DataFileModel'
+import AssessDayDataService from '../../services/AssessmentDayDataService'
+import _ from 'lodash'
+
+const AssessmentDayDataController = {
+  create: async (req, res) => {
+    try {
+      if (_.isEmpty(req.body))
+        return res.status(400).json({ message: 'The request is empty' })
+
+      const { appDb } = req.app.locals
+      const newAssessmentData = []
+      const updatedData = []
+      const assessmentDataService = new AssessDayDataService()
+
+      for await (const { metadata, subject_assessments } of req.body) {
+        if (metadata && subject_assessments.length) {
+          const { assessment, subject } = metadata
+          const query = {
+            assessment,
+            subject,
+          }
+          const storedAssessmentData = await AssessmentDayDataModel.all(
+            appDb,
+            query
+          )
+
+          newAssessmentData.push(
+            ...assessmentDataService.filterNewData(
+              subject_assessments,
+              storedAssessmentData
+            )
+          )
+          updatedData.push(
+            ...assessmentDataService.filterUpdatedData(
+              subject_assessments,
+              storedAssessmentData
+            )
+          )
+
+          await DataFileModel.update(appDb, query, metadata)
+        }
+      }
+
+      for await (const participantData of updatedData) {
+        const { assessment, subject, day } = participantData
+
+        await AssessmentDayDataModel.update(
+          appDb,
+          { assessment, subject, day },
```

Comment:
Do we ever get multiple assessments within the same day?

Reply:
There are multiple assessments that take place on the same day. 
The files are structured like this.
Day 1
Participant underwent three assessments. 
Results will be three different csv files one for each assessment